### ISSUE TITLE
Fetching search results asynchronously

### ISSUE THREAD

**Issue by olivierobert** [dongockhanh3103/nimble-tech-test#3]
https://github.com/dongockhanh3103/nimble-tech-test/blob/10cc1e0fc510a31b7bf821947c7ccb9cb4134b9e/app/controllers/attachment_controller.rb#L8-L17

The attachment controller calls synchronously the operations to fetch results. But in a real work application, we would not want the user to wait for all keywords to be searched. Imagine an upload with 100 keywords, it would take definitely more than a few seconds.

Also, it would be better to create a separate job per keyword search. As per the requirements, the search results should be collected for each keyword, not per upload.

**Comment by dongockhanh3103** [comment]
You are right Olivier. So that in this case I have 2 solutions for this one.
Firstly, I will implement to search by keyword that will work parallel and the gem I want to inject is [](https://github.com/grosser/parallel) . It is going to separate the keyword to individual thread and push the speed of search more. But this solution depend on the core of your server so that if the server has not much core it will not good for this case.

Secondly, my idea is migration the job of each keyword to background job and the gem that I have chosen is sidekiq. Sidekiq is familiar with each ruby on rails developer and easy to user.  Each keyword was moved to a job and work separately, and the prove is the application work faster. User does not wait for anything, all of this will run on the background job.

If you have any concern or good solution for this case. Lets me know.
